File: Medium/1004. Max Consecutive Ones III/solution.java
class Solution {
    public int longestOnes(int[] A, int K) {
        int zero = 0, left = 0, right = 0;
        int ans = 0;
        for (; right < A.length; right++) {
            if (A[right] == 0) {
                zero++;
            }
            while (zero > K) {
                if (A[left] == 0) {
                    zero--;
                }
                left++;
            }
            ans = Math.max(ans, right - left + 1);
        }
        return ans;
    }
}

File: Medium/1004. Max Consecutive Ones III/solution.py
class Solution:
    def longestOnes(self, nums: List[int], k: int) -> int:
        
        answer: int = 0

        left: int = 0
        count: int = 0

        for right in range(len(nums)):
            num = nums[right]
            if num == 0:
                count += 1
            
            while count > k:
                if nums[left] == 0:
                    count -= 1
                left += 1
            
            answer = max(answer, right - left + 1)
        
        return answer